functable.c

```diff
@@ -3,6 +3,10 @@
  * For conditions of distribution and use, see copyright notice in zlib.h
  */
 
+#if defined(_MSC_VER)
+#  include <windows.h>
```

Comment:
Including `windows.h` without limiting what other headers it can include will slow down compiling a lot. At minimum, `WIN32_LEAN_AND_MEAN` and `NOMINMAX` should be defined before including `windows.h`. It should also be the very last header to be included to avoid issues where subsequent headers try to define same macros with incompatible contents. This includes for example `MIN` and `MAX` macros that are normally not defined by default in C code, but are defined in zlib-ng.

Reply:
Done